builder/azure/arm/step_publish_to_shared_image_gallery.go

```diff
@@ -172,18 +182,23 @@ func (s *StepPublishToSharedImageGallery) Run(ctx context.Context, stateBag mult
 		miSigReplicaCount = constants.SharedImageGalleryImageVersionDefaultMaxReplicaCount
 	}
 
+	buildDiskEncryptionSetId := stateBag.Get(constants.ArmBuildDiskEncryptionSetId).(string)
```

Comment:
Is this attribute always set? Otherwise I'm slightly worried this will crash on the type cast

Reply:
Oops! It's not, and it does crash, good catch! I've addressed this in the newest commit